File: targets/jsts-web/src/vulns/vuln05_cmd.ts
import { Router } from "express";
import { execSync } from "child_process";
export const router = Router();
router.get("/vuln05/ping", (req, res) => {
  const host = String(req.query.host || "");
  // VULN:VULN-05:CWE-78:taint OS command injection — host concatenated into a shell command
  try { res.send(execSync("ping -c 1 " + host).toString()); } catch (e: any) { res.send(String(e.stdout || e.message)); }
});

File: targets/python-web/vulns/vuln05_cmd.py
import subprocess
from flask import Blueprint, request
bp = Blueprint("vuln05", __name__)

@bp.get("/vuln05/ping")
def ping():
    host = request.args.get("host", "")
    # VULN:VULN-05:CWE-78:taint OS command injection — host concatenated into a shell command
    return subprocess.getoutput("ping -c 1 " + host)
